Automattic/wp-calypso
# Reload cart when PopoverCart and StaleCartItemsNotice load

#### Changes proposed in this Pull Request

When a purchase completes in composite checkout, the transaction endpoint on the server will empty the cart, but if we're immediately redirected to a page in calypso which displays a cart notification (eg: the Plans page), it will still think there's products in the cart since the server can't notify us of the change. 

![Screen Shot 2020-05-15 at 3 50 53 PM](https://user-images.githubusercontent.com/2036909/82090612-e734ae80-96c3-11ea-9179-46f07c60ed74.png)

The proper solution to this is to stop using the CartStore and instead query the cart when it's needed. Eventually the CartStore can be deprecated and this problem will go away, but until then we need to help out.

This PR modifies the "cart" icon (the `PopoverCart` component) so that when it mounts it always fetches the cart. That way it should get an updated cart from the server when it is first loaded even if the CartStore's local cache is incorrect.

![Screen Shot 2020-05-15 at 3 51 03 PM](https://user-images.githubusercontent.com/2036909/82090621-eac83580-96c3-11ea-8ef0-c617fe58a961.png)

This PR also modifies the `StaleCartItemsNotice` with the same logic: fetch the cart when it first mounts and don't show the notice (which is the "Your cart is awaiting payment" message) if the cart is currently loading.

#### Testing instructions

I'm actually not 100% sure how to trigger the bug in the first place, although I know it happens. Suggestions welcome. I suppose it looks something like:

- Add a product to your cart and visit composite checkout.
- Complete a purchase using any payment method.
- Immediately visit the plans page and verify that the cart icon in the upper-right displays nothing in the cart.
